search/django/weedid/views.py

```diff
@@ -56,6 +57,27 @@ def upload_image(request):
         return HttpResponse("Only support POST request")
 
 
+def upload_agcontexts(request):
+    if request.method == "POST":
+        user_id = request.user.id
+        data = json.loads(request.body)
+        upload_id, ag_contexts = data["upload_id"], data["ag_contexts"]
+        weedcoco_path = os.path.join(
+            UPLOAD_DIR, str(user_id), str(upload_id), "weedcoco.json"
+        )
+        try:
+            add_agcontexts(weedcoco_path, ag_contexts)
+            Dataset.objects.filter(upload_id=upload_id).update(agcontext=[ag_contexts])
+        except Exception:
+            return HttpResponseNotAllowed("Failed to add AgContexts")
```

Comment:
We need to log the error that is raised here, and maybe send a greater error message to the frontend.

Reply:
Yeah I'm managing to implement exceptions for all scenarios